Parser/Token.cs

```diff
@@ -37,9 +37,24 @@ public enum E {
 
    // Utility function used to echo an error to the console
    public void PrintError () {
+      Console.OutputEncoding = Encoding.Unicode;
       if (Kind != ERROR) throw new Exception ("PrintError called on a non-error token");
       Console.ForegroundColor = ConsoleColor.Yellow;
-      Console.WriteLine ($"At line {Line}, column {Column} of {Source.FileName}: {Text}");
+      Console.Write ($"File: {Source.FileName}");
+      var cPos = Console.CursorLeft;
+      Console.WriteLine ($"\n\u2500\u2500\u2500\u252c{string.Join ("", Enumerable.Repeat ("\u2500", cPos - 3))}");
+      var lines = Source.Lines;
+      if (Line < lines.Length - 3) Console.WriteLine ($"{Line - 2,3}\u2502{lines[Line - 3]}");

```

Comment:
This repeated logic (Line < lines.Length - _n_) is a bit clumsy. Try to generalize it with a loop of some kind. In general, we want to be able to easily customize it to include a different number of lines of context. A helper routine like `int Clamp (this int n, int min, int max)` could be useful.

Reply:
Added helper functions